### ISSUE TITLE
Xcode ignores warnings reported by Swiftlint

### ISSUE THREAD

**Issue by PhilipDukhov** [realm/SwiftLint#3987]
### New Issue Checklist
- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)
- It may be a duplicate of #3757, but I can reproduce it without custom rules

### Describe the bug

Seems like Xcode doesn't display some repetitive warnings. I've created a list of same looking files:

```
for i in {1..10}; do
    echo "

import SwiftUI

class SomeClassNumber$i {
    
}

    " > State$i.swift
done
```

I'm using a default swiftlint file, and expect `trailing_whitespace`, `trailing_whitespace ` and `leading_whitespace` warnings to be displayed for each file.

swiftlint output is correct, but Xcode only highlights warnings in a random single file(or two of them) - a new one on each build.

<img width="1058" alt="Screenshot 2022-06-03 at 09 33 42" src="https://user-images.githubusercontent.com/6103621/171775473-a06d6058-f08d-46e1-8a02-ed455c9edf6b.png">

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files in current working directory
Linting 'State8.swift' (1/12)
Linting 'State3.swift' (2/12)
Linting 'State7.swift' (3/12)
Linting 'State5.swift' (4/12)
Linting 'State1.swift' (5/12)
Linting 'SwiftlintBugApp.swift' (6/12)
Linting 'State2.swift' (7/12)
Linting 'State9.swift' (8/12)
Linting 'State10.swift' (9/12)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State5.swift:1:1: warning: Leading Whitespace Violation: File shouldn't start with whitespace: currently starts with 2 whitespace characters (leading_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State5.swift:5:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State5.swift:2:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State5.swift:5:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State9.swift:1:1: warning: Leading Whitespace Violation: File shouldn't start with whitespace: currently starts with 2 whitespace characters (leading_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State9.swift:5:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State9.swift:2:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State9.swift:5:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State1.swift:1:1: warning: Leading Whitespace Violation: File shouldn't start with whitespace: currently starts with 2 whitespace characters (leading_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State1.swift:5:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State1.swift:2:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State1.swift:5:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
Linting 'State4.swift' (10/12)
Linting 'ContentView.swift' (11/12)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State7.swift:1:1: warning: Leading Whitespace Violation: File shouldn't start with whitespace: currently starts with 2 whitespace characters (leading_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State7.swift:5:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State7.swift:2:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State7.swift:5:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
Linting 'State6.swift' (12/12)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State2.swift:1:1: warning: Leading Whitespace Violation: File shouldn't start with whitespace: currently starts with 2 whitespace characters (leading_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State2.swift:5:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State2.swift:2:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State2.swift:5:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State3.swift:1:1: warning: Leading Whitespace Violation: File shouldn't start with whitespace: currently starts with 2 whitespace characters (leading_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State3.swift:5:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State3.swift:2:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State3.swift:5:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State8.swift:1:1: warning: Leading Whitespace Violation: File shouldn't start with whitespace: currently starts with 2 whitespace characters (leading_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State8.swift:5:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State8.swift:2:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State8.swift:5:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State4.swift:1:1: warning: Leading Whitespace Violation: File shouldn't start with whitespace: currently starts with 2 whitespace characters (leading_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State4.swift:5:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State4.swift:2:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State4.swift:5:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State10.swift:1:1: warning: Leading Whitespace Violation: File shouldn't start with whitespace: currently starts with 2 whitespace characters (leading_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State10.swift:5:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State10.swift:2:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State10.swift:5:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State6.swift:1:1: warning: Leading Whitespace Violation: File shouldn't start with whitespace: currently starts with 2 whitespace characters (leading_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State6.swift:5:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State6.swift:2:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
/Users/phil/Downloads/SwiftlintBug/SwiftlintBug/State6.swift:5:1: warning: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
Done linting! Found 40 violations, 0 serious in 12 files.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? **0.47.1**
* Installation method used (Homebrew, CocoaPods, building from source, etc)? **Homebrew**
* Paste your configuration file: **default file**
* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? **No**
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode **13.4**, Build version **13F17a**
* Do you have a sample that shows the issue? **No**, it's Xcode integration bug

Sample project: [SwiftlintBug.zip](https://github.com/realm/SwiftLint/files/8829088/SwiftlintBug.zip)

**Comment by marcelofabri** [comment]
Have you reported a bug with Apple? Since the SwiftLint output is correct, I'm afraid there's not much else we can do here. 

PS: I've also seen this before myself.

**Comment by marcelofabri** [comment]
Ideally, it'd be great if you could isolate this from SwiftLint. A shell script with a bunch of `echo` commands with [this format](https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Reporters/XcodeReporter.swift#L22) should be enough. IIRC whenever this issue happened to me was because more than one file got the same warning message on the same line number, so that could be a starting point.

Otherwise, I'd just list "install SwiftLint as according to documentation [link], eg with `brew install swiftlint`.

**Comment by PhilipDukhov** [comment]
Seems like it has being fixed in Xcode 14.0.0-Beta
